core/store.go

```diff
@@ -15,57 +15,58 @@ type WatchEvent struct {
 	Value     *Obj
 }
 
-var (
-	store     map[unsafe.Pointer]*Obj
-	expires   map[*Obj]uint64 // Does not need to be thread-safe as it is only accessed by a single thread.
-	keypool   map[string]unsafe.Pointer
-	WatchList sync.Map // Maps queries to the file descriptors of clients that are watching them.
-)
+type Store struct {
+	store        map[unsafe.Pointer]*Obj
+	expires      map[*Obj]uint64 // Does not need to be thread-safe as it is only accessed by a single thread.
+	keypool      map[string]unsafe.Pointer
+	storeMutex   sync.RWMutex
+	keypoolMutex sync.RWMutex
+	WatchList    sync.Map // Maps queries to the file descriptors of clients that are watching them.
 
-var (
-	storeMutex   sync.RWMutex // Mutex to protect the store map, must be acquired before keypoolMutex if both are needed.
-	keypoolMutex sync.RWMutex // Mutex to protect the keypool map, must be acquired after storeMutex if both are needed.
-)
+}
 
 // WatchChannel Channel to receive updates about keys that are being watched.
-// The Watcher goroutine will wait on this channel. When a key is updated, the
-// goroutine will send the updated value and the related operation to all the
-// clients that are watching the key.
 var WatchChannel chan WatchEvent
 
-func init() {
-	ResetStore()
-}
-
-func ResetStore() {
-	withLocks(func() {
-		store = make(map[unsafe.Pointer]*Obj)
-		expires = make(map[*Obj]uint64)
-		keypool = make(map[string]unsafe.Pointer)
-		WatchChannel = make(chan WatchEvent, config.KeysLimit)
-	}, WithStoreLock(), WithKeypoolLock())
+func NewStore() *Store {
+	s := &Store{
+		store:   make(map[unsafe.Pointer]*Obj),
+		expires: make(map[*Obj]uint64),
+		keypool: make(map[string]unsafe.Pointer),
+	}
+	WatchChannel = make(chan WatchEvent, config.KeysLimit)
+	return s
 }
 
-func NewObj(value interface{}, expDurationMs int64, oType, oEnc uint8) *Obj {
+func (store *Store) NewObj(value interface{}, expDurationMs int64, oType, oEnc uint8) *Obj {
 	obj := &Obj{
 		Value:          value,
 		TypeEncoding:   oType | oEnc,
 		LastAccessedAt: getCurrentClock(),
 	}
 	if expDurationMs >= 0 {
-		setExpiry(obj, expDurationMs)
+		store.setExpiry(obj, expDurationMs)
 	}
 	return obj
 }
 
+func (store *Store) ResetStore() {
```

Comment:
Do we need a locking mechanism for the resetStore function? I think we can take all values inside store to initial stage without acquiring locks.

Reply:
Would we see any inconsistencies when clients read which the reset is still in progress. Eg. Some aributes are reset but others are not?

Reply:
I believe resetStore is only for tests and it doesn't get involved in any client operations. 
<img width="332" alt="Screenshot 2024-08-18 at 4 17 05 PM" src="https://github.com/user-attachments/assets/efc13006-c941-4964-9b68-ca4dc5c468aa">

Reply:
Good find. Makes me think then should it even in the store.go. Seems more like a utility func for tests.